src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/StandardChunkVisitor.java

```diff
@@ -0,0 +1,64 @@
+package org.jenkinsci.plugins.workflow.graphanalysis;
+
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * Simple handler for linear chunks (basic stages, etc), designed to be extended
+ * Note: does not handle parallels or nesting
```

Comment:
Can you expand on what concretely it does not handle, and how it behaves if given graphs with such structures?